retrieval/scrape_test.go

```diff
@@ -222,54 +222,42 @@ func TestScrapePoolReload(t *testing.T) {
 	}
 }
 
-func TestScrapePoolReportAppender(t *testing.T) {
+func TestScrapeLoopWrapSampleAppender(t *testing.T) {
 	cfg := &config.ScrapeConfig{
 		MetricRelabelConfigs: []*config.RelabelConfig{
-			{}, {}, {},
+			{
+				Action:       config.RelabelDrop,
+				SourceLabels: model.LabelNames{"__name__"},
+				Regex:        config.MustNewRegexp("a.*"),
+			},
+			{
+				Action:       config.RelabelDrop,
+				SourceLabels: model.LabelNames{"__name__"},
+				Regex:        config.MustNewRegexp("a.*"),
+			},
 		},
 	}
+
 	target := newTestTarget("example.com:80", 10*time.Millisecond, nil)
 	app := &nopAppender{}
 
 	sp := newScrapePool(context.Background(), cfg, app)
 
 	cfg.HonorLabels = false
-	wrapped := sp.reportAppender(target)
 
-	rl, ok := wrapped.(ruleLabelsAppender)
-	if !ok {
-		t.Fatalf("Expected ruleLabelsAppender but got %T", wrapped)
-	}
-	if rl.SampleAppender != app {
-		t.Fatalf("Expected base appender but got %T", rl.SampleAppender)
-	}
+	sl := sp.newLoop(
+		sp.ctx,
+		&targetScraper{Target: target, client: sp.client},
+		sp.appender,
+		target.Labels(),
+		sp.config,
+	).(*scrapeLoop)
+	wrapped, counting := sl.wrapAppender(sl.appender)
+	wrapped.Append(&model.Sample{})
 
-	cfg.HonorLabels = true
-	wrapped = sp.reportAppender(target)
-
-	hl, ok := wrapped.(ruleLabelsAppender)
-	if !ok {
-		t.Fatalf("Expected ruleLabelsAppender but got %T", wrapped)
+	if counting.count != 1 {
```

Comment:
I'm appending one sample. This tests if the counting appender is counting it.

("Works for me", locally... not sure why travis doesn't like this line.)